vendor/purs/runtime/irregex/irregex.scm

```diff
@@ -0,0 +1,3984 @@
+;;;; irregex.scm -- IrRegular Expressions
+;;
+;; Copyright (c) 2005-2021 Alex Shinn.  All rights reserved.
+;; BSD-style license: http://synthcode.com/license.txt
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; At this moment there was a loud ring at the bell, and I could
+;; hear Mrs. Hudson, our landlady, raising her voice in a wail of
+;; expostulation and dismay.
+;;
+;; "By heaven, Holmes," I said, half rising, "I believe that
+;; they are really after us."
+;;
+;; "No, it's not quite so bad as that.  It is the unofficial
+;; force, -- the Baker Street irregulars."
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;; Notes
+;;
+;; This code should not require any porting - it should work out of
+;; the box in any R[457]RS Scheme implementation.  Slight modifications
+;; are needed for R6RS (a separate R6RS-compatible version is included
+;; in the distribution as irregex-r6rs.scm).
```

Comment:
Do we need to include `irregex-r6rs.scm` as well?

Reply:
> Do we need to include `irregex-r6rs.scm` as well?

According to [Community Scheme Wiki](http://community.schemewiki.org/?Scheme-FAQ-Standards) Chez complies with R5RS. I'm not versed in Scheme standards though.

Reply:
Latest CSUG [claims support for R5RS](https://cisco.github.io/ChezScheme/csug9.6/libraries.html#./libraries:h1).

Reply:
That link talks about R6RS though? 🤔

Reply:
There's `(rnrs r5rs (6))` in the list of built-in libraries.

Reply:
I guess we could try it out this way and see if it works fine